hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -262,9 +268,16 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
     this.instantState.clear();
     if (this.issuedInstant != null) {
       this.instantState.add(this.issuedInstant);
+      this.readMetrics.setIssuedInstant(this.issuedInstant);
     }
     if (this.issuedOffset != null) {
```

Comment:
Does the metrics got updated for each read?

Reply:
The metrics will be updated for each checkpoint